src/Confluent.Kafka/Producer.cs

```diff
@@ -218,6 +231,7 @@ private void DeliveryReportCallbackImpl(IntPtr rk, IntPtr rkmessage, IntPtr opaq
                 }
             }
 
+            ErrorCode err;
```

Comment:
sorry there's a bit of noise here. it's slightly better, but unrelated to the rest of this pr as it turned out.